core/src/main/scala/org/apache/spark/deploy/history/EventLogFileWriters.scala

```diff
@@ -0,0 +1,420 @@
+  private def rollEventLogFile(): Unit = {
+    writer.foreach(_.close())
+
+    index += 1
+    currentEventLogFilePath = getEventLogFilePath(logDirForAppPath, appId, appAttemptId, index,
+      compressionCodecName)
+
+    val (hadoopStream, outputStream) = initLogFile(currentEventLogFilePath)
+    hadoopDataStream = hadoopStream
+    countingOutputStream = Some(new CountingOutputStream(outputStream))
+    writer = Some(new PrintWriter(
+      new OutputStreamWriter(countingOutputStream.get, StandardCharsets.UTF_8)))
+  }
+
+  override def stop(): Unit = {
+    writer.foreach(_.close())
+    val appStatusPathIncomplete = getAppStatusFilePath(logDirForAppPath, appId, appAttemptId,
+      inProgress = true)
+    val appStatusPathComplete = getAppStatusFilePath(logDirForAppPath, appId, appAttemptId,
+      inProgress = false)
+    renameFile(appStatusPathIncomplete, appStatusPathComplete, overwrite = true)
+  }
+
+  override def logPath: String = logDirForAppPath.toString
+
+  private def createAppStatusFile(inProgress: Boolean): Unit = {
+    val appStatusPath = getAppStatusFilePath(logDirForAppPath, appId, appAttemptId, inProgress)
+    val outputStream = fileSystem.create(appStatusPath)
+    // we intentionally create zero-byte file to minimize the cost
+    outputStream.close()
+  }
+}
+
+object RollingEventLogFilesWriter {
+  def getAppEventLogDirPath(logBaseDir: URI, appId: String, appAttemptId: Option[String]): Path =
+    new Path(new Path(logBaseDir), "eventlog_v2_" +
+      EventLogFileWriter.nameForAppAndAttempt(appId, appAttemptId))
+
+  def getAppStatusFilePath(
+      appLogDir: Path,
+      appId: String,
+      appAttemptId: Option[String],
+      inProgress: Boolean): Path = {
+    val base = "appstatus_" + EventLogFileWriter.nameForAppAndAttempt(appId, appAttemptId)
+    val name = if (inProgress) base + EventLogFileWriter.IN_PROGRESS else base
+    new Path(appLogDir, name)
+  }
+
+  def getEventLogFilePath(
+      appLogDir: Path,
+      appId: String,
+      appAttemptId: Option[String],
+      index: Long,
+      codecName: Option[String]): Path = {
+    val base = s"events_${index}_" + EventLogFileWriter.nameForAppAndAttempt(appId, appAttemptId)
+    val codec = codecName.map("." + _).getOrElse("")
+    new Path(appLogDir, base + codec)
+  }
+
+  def isEventLogDir(status: FileStatus): Boolean = {
+    status.isDirectory && status.getPath.getName.startsWith("eventlog_v2_")
+  }
+
+  def isEventLogFile(status: FileStatus): Boolean = {
+    status.isFile && isEventLogFile(status.getPath)
+  }
+
+  def isEventLogFile(path: Path): Boolean = {
+    path.getName.startsWith("events_")
+  }
+
+  def isAppStatusFile(status: FileStatus): Boolean = {
+    status.isFile && isAppStatusFile(status.getPath)
+  }
+
+  def isAppStatusFile(path: Path): Boolean = {
+    path.getName.startsWith("appstatus")
+  }
+
+  def getSequence(eventLogFileName: String): Long = {
+    require(eventLogFileName.startsWith("events_"), "Not a event log file!")
```

Comment:
Using constant variable for such prefix ?

Reply:
Nice suggestion! Will address.